kafka-ssl/README.md

```diff
@@ -0,0 +1,133 @@
+# Debezium - Kafka with SSL example
+
+This demo automatically deploys the topology of services as defined in the [Debezium Tutorial](https://debezium.io/documentation/reference/stable/tutorial.html) with Kafka SSL enabled.
+
+## Configuration
+
+The kafka instance will be deployed using the certificates at the './resources' folder:
+```yaml
+  kafka:
+    image: quay.io/debezium/kafka:${DEBEZIUM_VERSION}
+    ports:
+     - "9092:9092"
+    links:
+     - zookeeper
+    environment:
+     - ZOOKEEPER_CONNECT=zookeeper:2181
+     - KAFKA_LISTENER_SECURITY_PROTOCOL_MAP=BROKER:PLAINTEXT,SSL:SSL,CONTROLLER:PLAINTEXT
+     - KAFKA_LISTENERS=BROKER://0.0.0.0:9093,SSL://0.0.0.0:9092,CONTROLLER://0.0.0.0:9094
+     - KAFKA_ADVERTISED_LISTENERS=SSL://localhost:9092,BROKER://localhost:9093
+     - KAFKA_SSL_CLIENT_AUTH=required
+     - KAFKA_SSL_KEYSTORE_LOCATION=/opt/config/ssl/kafka-ssl-keystore.p12
+     - KAFKA_SSL_KEYSTORE_PASSWORD=top-secret
+     - KAFKA_SSL_KEYSTORE_TYPE=PKCS12
+     - KAFKA_SSL_TRUSTSTORE_LOCATION=/opt/config/ssl/kafka-ssl-truststore.p12
+     - KAFKA_SSL_TRUSTSTORE_PASSWORD=top-secret
+     - KAFKA_INTER_BROKER_LISTENER_NAME=BROKER
+    volumes:
+      - ./resources:/opt/config/ssl:z
+```
+
+The certificates will be mounted at the folder "/opt/config/ssl".
+
+Then, when configuring the debezium connector, we need to also mount these certificates and properly configure the connector:
+
+```yaml
+  connect:
+    image: quay.io/debezium/connect:${DEBEZIUM_VERSION}
+    ports:
+      - "8083:8083"
+    environment:
+      - BOOTSTRAP_SERVERS=localhost:9092
+      - GROUP_ID=1
+      - CONFIG_STORAGE_TOPIC=my_connect_configs
+      - OFFSET_STORAGE_TOPIC=my_connect_offsets
+      - STATUS_STORAGE_TOPIC=my_connect_statuses
+      - CONNECT_SECURITY_PROTOCOL=SSL
+      - CONNECT_SSL_KEYSTORE_LOCATION=/opt/config/ssl/kafka-ssl-keystore.p12
+      - CONNECT_SSL_KEYSTORE_PASSWORD=top-secret
+      - CONNECT_SSL_KEYSTORE_TYPE=PKCS12
+      - CONNECT_KEY_PASSWORD=top-secret
+      - CONNECT_SSL_TRUSTSTORE_LOCATION=/opt/config/ssl/kafka-ssl-truststore.p12
+      - CONNECT_SSL_TRUSTSTORE_PASSWORD=top-secret
+      - CONNECT_SSL_TRUSTSTORE_TYPE=PKCS12
+    network_mode: host
```

Comment:
Why is this necessary?

Reply:
Because the certificate I use validates that the host is localhost. We can configure Kafka to disable the hostname validation as well.

Reply:
Intersting, I am getting `Error response from daemon: conflicting options: host type networking can't be used with links. This would result in undefined behavior` when I try it with the option, but maybe we can get rid of it and have a certificate generated for hostname `kafka`?

Reply:
In that case I believe your networking and advertisment setup will simplify a lot too.